beer-garden/functionary
# Establish basic unit test structure and some examples

Closes #88 

This PR establishes some basic structure for unit tests, with a smattering of examples.

* For the django project, tests are housed in the individual app directories and mirror the structure of the app itself
* cli has a top level tests folder. It contains a single test, and a conftest.py with a couple of fixtures that will likely be generally useful.
* No tests have been created for the runner code yet.
* A github action has been added to run the tests on PRs.
* The vscode settings have been updated to enable pytest.  The tests should be auto-discovered, but if they aren't just do the basic test config and tell it you are using pytest.

Other notes:
* I slightly re-organized the settings files, converting base.py into its own module.  This was so that the environment configs (debug, test, prod) are more clearly distinguishable from the base config files.

## Testing Instructions
* Be sure to pip install the new dependencies, specifically the requirements files in the cli and functionary folders.
* From the top level directory you can run:
  * pytest cli
  * pytest functionary
* You should also make sure you can use the testing component in vscode to run the tests.
